File: app/src/main/java/com/lenar/fitforlife/utils/SharedPreferencesHelper.java
package com.lenar.fitforlife.utils;

import android.app.Application;
import android.content.SharedPreferences;

import com.lenar.fitforlife.FitForLifeApplication;

import static android.content.Context.MODE_PRIVATE;

public class SharedPreferencesHelper {

    private static final String SETTING_PREFERENCES_FILE = "com.alekzunder.weatherapp.settings";

    public static final String KEY_KEEP_SCREEN_ON = "com.alekzunder.weatherapp.keep_screen_on";
    public static final String KEY_UPDATING_TIME = "com.alekzunder.weatherapp.updating_time";
    public static final String KEY_UPDATE_AUTOMATICALLY = "com.alekzunder.weatherapp.update_automatically";

    private static final String DEFAULT_STRING = "";
    private static final int DEFAULT_INT = 30;
    private static final boolean DEFAULT_BOOLEAN = false;

    private Application mApplication;

    public SharedPreferencesHelper(FitForLifeApplication application) {
        mApplication = application;
    }

    private SharedPreferences getSPrefs() {
        return mApplication.getSharedPreferences(SETTING_PREFERENCES_FILE, MODE_PRIVATE);
    }

    public void saveBoolean(String key, boolean value) {
        getSPrefs().edit().putBoolean(key, value).apply();
    }

    public boolean getBoolean(String key) {
        return getSPrefs().getBoolean(key, DEFAULT_BOOLEAN);
    }

    public void saveString(String key, String value) {
        getSPrefs().edit().putString(key, value).apply();
    }

    public String getString(String key) {
        return getSPrefs().getString(key, DEFAULT_STRING);
    }

    public void saveInt(String key, int value) {
        getSPrefs().edit().putInt(key, value).apply();
    }

    public int getInt(String key) {
        return getSPrefs().getInt(key, DEFAULT_INT);
    }
}

File: app/src/main/java/com/nilden/fitforlife/utils/SharedPreferencesHelper.kt
package com.nilden.fitforlife.utils

import android.app.Application
import android.content.Context.MODE_PRIVATE
import android.content.SharedPreferences
import com.nilden.fitforlife.FitForLifeApplication

class SharedPreferencesHelper(application: FitForLifeApplication) {

    private val mApplication: Application

    private val sPrefs: SharedPreferences
        get() = mApplication.getSharedPreferences(SETTING_PREFERENCES_FILE, MODE_PRIVATE)

    init {
        mApplication = application
    }

    fun saveBoolean(key: String, value: Boolean) {
        sPrefs.edit().putBoolean(key, value).apply()
    }

    fun getBoolean(key: String): Boolean {
        return sPrefs.getBoolean(key, DEFAULT_BOOLEAN)
    }

    fun getString(key: String): String {
        return sPrefs.getString(key, DEFAULT_STRING)
    }

    fun saveInt(key: String, value: Int) {
        sPrefs.edit().putInt(key, value).apply()
    }

    fun getInt(key: String): Int {
        return sPrefs.getInt(key, DEFAULT_INT)
    }

    companion object {

        private const val SETTING_PREFERENCES_FILE = "com.nilden.firforlife.settings"

        const val KEY_KEEP_SCREEN_ON = "com.nilden.firforlife.keep_screen_on"
        const val KEY_UPDATING_TIME = "com.nilden.firforlife.updating_time"
        const val KEY_UPDATE_AUTOMATICALLY = "com.nilden.firforlife.update_automatically"

        private const val DEFAULT_STRING = ""
        private const val DEFAULT_INT = 30
        private const val DEFAULT_BOOLEAN = false
    }
}
